Mathlib/Algebra/Group/Submonoid/BigOperators.lean

```diff
@@ -138,4 +141,23 @@ theorem noncommProd_mem (S : Submonoid M) {ι : Type*} (t : Finset ι) (f : ι 
   rintro ⟨x, ⟨hx, rfl⟩⟩
   exact h x hx
 
+end Monoid
+
+section CommMonoid
+variable [CommMonoid M] {s : Finset M} {x : M}
+
+@[to_additive]
+lemma mem_closure_finset : x ∈ closure s ↔ ∃ n : M → ℕ, x = ∏ a ∈ s, a ^ n a where
```

Comment:
Can you strengthen this to say n is only non-one in s?

Reply:
My preference is to keep it from `M` and have a propositional part for this, to avoid coercing from a subtype: the product is only over `s` anyway.